src/Mobs/Skeleton.cpp

```diff
@@ -63,6 +63,28 @@ bool cSkeleton::Attack(std::chrono::milliseconds a_Dt)
 
 
 
+void cSkeleton::KilledBy(TakeDamageInfo & a_TDI)
+{
+	Super::KilledBy(a_TDI);
+
+	if ((a_TDI.Attacker != nullptr) && (a_TDI.Attacker->IsPlayer()) && (a_TDI.DamageType == dtArrowAttack))
+	{
+		cPlayer * Player = static_cast<cPlayer *>(a_TDI.Attacker);
+
+		int DistX = FloorC(Player->GetPosX()) - FloorC(GetPosX());
+		int DistZ = FloorC(Player->GetPosZ()) - FloorC(GetPosZ());
+
+		if (DistX * DistX + DistZ * DistZ >= 2500)
+		{
+			Player->AwardAchievement(CustomStatistic::AchSnipeSkeleton);
+		}
+	}
```

Comment:
It could be better to move this to `cPlayer::Killed` where some achievement stuff is already at.
Is flooring the individual components then subtracting, instead of subtracting then flooring, intentional?

Reply:
I looked again and whilst the signature is `void cPlayer::Killed(cEntity * a_Victim)` you don't have access to the DamageType. You could modify the signature to pass it, that way Skeleton.cpp doesn't have to `#include Player.cpp`, but this may be too much trouble. Your choice.

Reply:
I'd like to remain all on the same places where it is. I think, specific achievements rewards, not general like kill any enemy should be placed where it logically should to avoid additional checks and make it more intuitive.